RFCs/0016-Loggers.md

```diff
@@ -0,0 +1,85 @@
+# 0016 Loggers
+
+## Summary
+This note outlines the proposed changes for:
+1. Enabling logger support in test platform for TranslationLayer clients.
+2. Allowing users to provide loggers from the runsettings.
+
+## Motivation
+1. Enabling logger support for TranslationLayer clients will allow clients to use logger extensibility feature.
+
+2. Allowing users to provide loggers from the runsettings enables:
+  * TranslationLayer clients to provide loggers to the test platform.
+  * Alternative way to provide loggers in addition to the existing command line logger argument.
+
+## Specifying a logger
+Loggers can be provided to the test platform in one of the following ways:
+
+1.  "/Logger:"UriOrFriendlyName";Key1=Value1;Key2=Value2" This is a switch to vstest.console.exe that feeds logger information to the test platform. For instance:
+
+    ```
+    /Logger:"logger://Microsoft/TestPlatform/TrxLogger/v1"
+    ```
+
+    This loads and initializes logger with "logger://Microsoft/TestPlatform/TrxLogger/v1" as uri.
+
+    ```
+    /Logger:sampleLogger;Key1=Value1;Key1=Value2
+    ```
+
+    This loads and initializes logger with "sampleLogger" as friendly name. Key1=Value1 and Key2=Value2 are passed as dictionary parameters to the logger while initialization.
+
+2. Runsettings via "Logger" node in the LoggerRunSettings section. Here is a sample on how this can be specified:
+
+    ```xml
+    <RunSettings>
+        <LoggerRunSettings>
+            <Loggers>
+                <Logger friendlyName="sampleLoggerwithParameters">
+                    <Configuration>
+                        <Key1>Value1</Key1>
+                        <Key2>Value2</Key2>
+                    </Configuration>
+                </Logger>
+                <Logger uri="logger://sample/sampleLoggerWithoutParameters1"
+                        friendlyName="sampleLoggerWithoutParameters1" />
+                <Logger uri="logger://sample/sampleLoggerWithoutParameters2"
+                        assemblyQualifiedName="Sample.Sample.Sample.SampleLogger, Sample.Sample.Logger, Version=0.0.0.0, Culture=neutral, PublicKeyToken=xxxxxxxxxxxxxxxx"
+                        friendlyName="sampleLoggerWithoutParameters2" />
+            </Loggers>
+        </LoggerRunSettings>
+    </RunSettings>
+    ```
+
+    This loads and initializes:
```

Comment:
Can you please specify the exact logger APIs and the parameters it will be invoked with (for the above runsettings sample)?

Reply:
Loggers from run settings mimics the behavior of loggers from command line (from logger initialization perspective).

I will still add logger API which will invoked while initialization in RFC. thanks.